### PR TITLE
Adding Guards 👮

### PR DESCRIPTION
**Pull request by AliSoftware**
This PR goes on top of #32 and convert the code to modern Swift by using `guard` to make it waaayy more readable :wink: 

I kept #32 separated from this PR as they are not directly related and you might want to review them separately, but this PR still requires #32 to be merged first — as the added `guard` statements also apply to changes made by #32

### CODE REVIEW

File: Stencil/Node.swift [hunk 1]
```diff
@@ -73,10 +73,13 @@ public class VariableNode : NodeType {
 public class NowNode : NodeType {
   public let format:Variable
 
-  public class func parse(parser:TokenParser, token:Token) -> NodeType {
+  public class func parse(parser:TokenParser, token:Token) throws -> NodeType {
     var format:Variable?
 
     let components = token.components()
+    guard components.count <= 2 else {
+      throw TemplateSyntaxError("'now' tags should can only have one argument: the format string `\(token.contents)`.")
```
Comment (kylef) [inline Stencil/Node.swift]: Please switch `can` to `may` (RFC 2119)

**Comment by kylef** [comment]
This PR still includes the changes for #32, can you remove these from this PR please.

File: Stencil/Node.swift [hunk 2]
```diff
@@ -118,33 +117,37 @@ public class ForNode : NodeType {
...
 
-    throw TemplateSyntaxError("'for' statements should use the following 'for x in y' `\(token.contents)`.")
+    return ForNode(variable: variable, loopVariable: loopVariable, nodes: forNodes, emptyNodes: emptyNodes)
   }
 
   public init(variable:String, loopVariable:String, nodes:[NodeType], emptyNodes:[NodeType]) {
     self.variable = Variable(variable)
     self.loopVariable = loopVariable
     self.nodes = nodes
+    // TODO: Handle emptyNodes
```
Comment (kylef) [inline Stencil/Node.swift]: Good spot

File: Stencil/TemplateLoader/Inheritence.swift [hunk 3]
```diff
@@ -58,22 +58,20 @@ class ExtendsNode : NodeType {
...
-
-      let paths:String = loader.paths.map { path in
-        return path.description
-        }.joinWithSeparator(", ")
+    guard let loader = context["loader"] as? TemplateLoader else {
+      throw TemplateSyntaxError("Template loader not in context")
+    }
+    
+    guard let template = loader.loadTemplate(templateName) else {
+      let paths:String = loader.paths.map { path in return path.description }.joinWithSeparator(", ")
```
Comment (kylef) [inline Stencil/TemplateLoader/Inheritence.swift]: `return` here is unnecessary since single line closures have implicit return

**Comment by AliSoftware** [comment]
> This PR still includes the changes for #32, can you remove these from this PR please.

Not really actually, as they would conflict if separated: there are `guard` statements added there that are on the same code regions that the ones where I added `parser.eatNextCRLF()` calls. Splitting those PRs would make the merge conflict much harder ^^

File: Stencil/Node.swift (hunk 2 above)
Comment (AliSoftware) [inline Stencil/Node.swift]: I wonder why the compiler didn't warn about unused variable.parameter btw, I'd expect it to suggest you a _Fix-It_ to replace `emptyNodes` with `_` in such cases.

**Comment by AliSoftware** [comment]
You can isolate [the diff between #32 and this PR here](https://github.com/AliSoftware/Stencil/compare/fix/22-extra_lines...AliSoftware:guards) if you want to review it more easily

**Comment by AliSoftware** [comment]
Ok, this is now entirely rewritten without the commits from #32 and ready to merge!

**Comment by kylef** [comment]
Thanks for the pull request!

**Pull request merged**